f5/bigip/sys/dns.py

```diff
@@ -0,0 +1,48 @@
+# coding=utf-8
+#
+# Copyright 2016 F5 Networks Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+"""BIG-IP® system dns module
+
+REST URI
+    ``http://localhost/mgmt/tm/sys/dns``
+
+GUI Path
+    ``System --> Configuration --> Device --> DNS``
+
+REST Kind
+    ``tm:sys:dns:*``
+"""
+
+from f5.bigip.mixins import UnnamedResourceMixin
+from f5.bigip.resource import Resource
+
+
+class Dns(UnnamedResourceMixin, Resource):
+    """BIG-IP® system DNS unnamed resource
+
+        .. note::
+
+        This is an unnamed resource so it has not ~Partition~Name pattern
+        at the end of its URI.
+    """
+    def __init__(self, sys):
+        super(Dns, self).__init__(sys)
+        endpoint = self.__class__.__name__.lower()
+        self._meta_data['required_load_parameters'] = set()
+        self._meta_data['required_json_kind'] = 'tm:sys:dns:dnsstate'
+        self._meta_data['uri'] = \
+            self._meta_data['container']._meta_data['uri'] + endpoint + '/'
```

Comment:
This logic needs to be pushed into the `UnnamedResourceMixin`, it is (or should be) a defining feature of that Mixin.    It's not a blocker on this PR, but if we do not migrate it in this PR, then we should make it a blocking issue for future implementations of `UnnamedResourceMixin` concrete subclasses.